compiler/rustc_session/src/lint/builtin.rs

```diff
@@ -553,6 +553,11 @@ declare_lint! {
         reference: "issue #76200 <https://github.com/rust-lang/rust/issues/76200>",
         edition: None,
     };
+
+declare_lint! {
+    pub FUNCTION_REFERENCES,
```

Comment:
The name might need some bikeshedding... maybe `FUNCTION_ITEM_REFERENCES` or so would be clearer? At least it is more precise.^^